controllers/etcdadmconfig_controller.go

```diff
@@ -99,7 +99,7 @@ func (r *EtcdadmConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl
 	}
 
 	err = c.Watch(
-		&source.Kind{Type: &clusterv1.Cluster{}},
+		source.Kind(mgr.GetCache(), &clusterv1.Cluster{}),
 		handler.EnqueueRequestsFromMapFunc(r.ClusterToEtcdadmConfigs),
 		predicates.ClusterUnpausedAndInfrastructureReady(r.Log),
```

Comment:
can this moved to the the main chain of setup with another `Watches`?
like the Machines